### DISCUSSION TITLE
New programmer looking for help setting up Kubernetes dev dashboard

### DISCUSSION THREAD

**Discussion by jlweikit** [kubernetes/dashboard#9418] [General]
Hi everyone,

I’m currently trying to set up a development environment for the Kubernetes Dashboard project, but I’ve run into an issue where the dev.serve.sh script seems to be missing. This script is referenced in the dev.web.Dockerfile file, but I can’t find it anywhere in the repository.

Here’s what I have:

Operating System: Ubuntu via WSL2 on Windows
Kubernetes Dashboard Repo: I'm working with the master branch
Error: When I run make serve, I get the following error:

vbnet

/bin/sh: ./hack/scripts/dev.serve.sh: Permission denied
Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error mounting "/run/desktop/mnt/host/wsl/docker-desktop-bind-mounts/Ubuntu/...": mount ...: no such file or directory: unknown
make: *** [Makefile:79: serve] Error 1

I’ve checked the ./hack/scripts/ directory, but the dev.serve.sh script doesn’t exist. I’ve also reviewed the project’s documentation and Makefiles, but there’s no mention of generating or configuring this script.

Questions:

Is this script supposed to be auto-generated during the setup, or should it be included in the repo?
If it's supposed to be generated, what step might I be missing that's causing it to not appear?
Has anyone else run into this issue? If so, how did you resolve it?

Any help or guidance would be greatly appreciated! Thank you!

**Comment by floreks** [comment]
This script is inside the module itself: https://github.com/kubernetes/dashboard/blob/master/modules/web/hack/scripts/dev.serve.sh
Docker compose ensures that context is set to a correct directory and mounts required directories that are then being watched for changes. It is important to run `make serve` from the repo root directory. Also, depending on how your docker is installed, it might not have permission to access mounted files.

As you can see there is a permission denied error in your log:
> /bin/sh: ./hack/scripts/dev.serve.sh: Permission denied
